RFC-007-SWAP-Ethereum-Basic.md

```diff
@@ -0,0 +1,291 @@
+    selfdestruct(<refund_identity>)
+}
+
+```
+
+The following is the contract deployment code encoded template as hex which will deploy the above contract when the placeholder values (set to `00`s) are replaced with the parameters.
+
+```
+6100dc61000f6000396100dc6000f336156051576020361415605c57602060006000376020602160206000600060026048f17f0000000000000000000000000000000000000000000000000000000000000000602151141660625760006000f35b42630000000010609f575b60006000f35b7fb8cac300e37f03ad332e581dea21b2f0b84eaaadc184a295fef71e81f44a741360206000a1730000000000000000000000000000000000000000ff5b7f5d26862916391bf49478b2f5103b0720a842b45ef145a268f2cd1fb2aed5517860006000a1730000000000000000000000000000000000000000
+```
+
+To compile the contract code replace the placeholder values with the HTLC paramters at the following offsets:
+
+
+| Data              | Position of first byte | Position of last byte | Length |
+|:------------------|:-----------------------|:----------------------|:-------|
+| `secret_hash`     | 51                     | 82                    | 32     |
+| `expiry`          | 99                     | 102                   | 4      |
+| `redeem_identity` | 153                    | 172                   | 20     |
+| `refund_identity` | 219                    | 248                   | 20     |
+
+The contract emits two logs:
+
+| Topic        | keccack256(Topic)                                                  | Data     | Emitted When             |
+|:-------------|:-------------------------------------------------------------------|----------|:-------------------------|
+| `Redeemed()` | 0xb8cac300e37f03ad332e581dea21b2f0b84eaaadc184a295fef71e81f44a7413 | `secret` | The contract is redeemed |
+| `Refunded()` | 0x5d26862916391bf49478b2f5103b0720a842b45ef145a268f2cd1fb2aed55178 | -        | The contract is refunded |
+
+
+Implementations SHOULD use the following gas limits on the transactions related to the contract:
+
+| Transaction | recommended gas limit |
+|:------------|-----------------------|
+| Deployment  | 121,000               |
+| Redeem      | 100,000               |
+| Refund      | 100,000               |
+
+## Execution Phase
+
+The following section describes how both parties should interact with the Ethereum blockchain during the [RFC003 execution phase](./RFC-003-SWAP-basic.md#execution-phase).
+
+### Deployment
+
+At the start of the deployment stage, both parties compile the contract code as described in the previous section.
+We will call this value `contract_code`.
+
+To deploy the Ether HTLC, the *funder* must deploy the `contract_code`.
+They SHOULD do this by sending a contract deployment transaction to the relevant Ethereum blockchain with:
+
+- `contract_code` as the `data` of the transaction
+- The quantity of wei specified in the Ether Asset header as the `value` of the transaction
+
+The funder SHOULD NOT do this executing the `CREATE` opcode from another contract.
```

Comment:
Can you clarify what you mean with the sentence?

Reply:
Note @bonomat made a suggestion clarifying this which I took.